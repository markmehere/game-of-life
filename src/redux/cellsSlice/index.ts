import { cellsSlice } from "./cellsSlice";

export default cellsSlice.reducer;
