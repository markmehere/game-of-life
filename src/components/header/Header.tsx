import { useAppDispatch } from "redux/hooks";
import styled from "styled-components";
import { next, reset } from "redux/cellsSlice/cellsSlice";

const Row = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 30rem;
  height: 4rem;
`;

const Button = styled.button`
  font-size: 1.2rem;
`;

export function Header() {
  const dispatch = useAppDispatch();

  return (
    <Row>
      <Button aria-label="Reset" onClick={() => dispatch(reset())}>
        Reset
      </Button>
      <Button aria-label="Next generation" onClick={() => dispatch(next())}>
        Next generation
      </Button>
    </Row>
  );
}
