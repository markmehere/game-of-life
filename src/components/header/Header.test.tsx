import { screen as s, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { store } from "redux/store";
import * as hooks from "redux/hooks";
import { Header } from "./Header";

describe("<Header />", () => {
  it("matches snapshot", async () => {
    const wrapper = render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    const nextButton = await s.findByRole("button", {
      name: "Next generation",
    });
    expect(nextButton).toBeInTheDocument();
    const resetButton = await s.findByRole("button", {
      name: "Reset",
    });
    expect(resetButton).toBeInTheDocument();
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it("reset button dispatches on click", async () => {
    const spy = jest.spyOn(hooks, "useAppDispatch");
    const mockDispatch = jest.fn();
    spy.mockReturnValue(mockDispatch);
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    const resetButton = await s.findByRole("button", {
      name: "Reset",
    });
    userEvent.click(resetButton);
    expect(mockDispatch).toHaveBeenCalled();
  });

  it("next button dispatches on click", async () => {
    const spy = jest.spyOn(hooks, "useAppDispatch");
    const mockDispatch = jest.fn();
    spy.mockReturnValue(mockDispatch);
    render(
      <Provider store={store}>
        <Header />
      </Provider>
    );
    const nextButton = await s.findByRole("button", {
      name: "Next generation",
    });
    userEvent.click(nextButton);
    expect(mockDispatch).toHaveBeenCalled();
  });
});
