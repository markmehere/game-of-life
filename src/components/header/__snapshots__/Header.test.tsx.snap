// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-bdvvtL hQzVqw"
    >
      <button
        aria-label="Reset"
        class="sc-gsDKAQ hYtlMt"
      >
        Reset
      </button>
      <button
        aria-label="Next generation"
        class="sc-gsDKAQ hYtlMt"
      >
        Next generation
      </button>
    </div>
  </div>
</body>
`;
